### PR TITLE
Hook for dynamic price

### PR DESCRIPTION
**Pull request by josemariagomezroncero**
Hook for dynamic price

# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "Fix", "Close" or "New" section*
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed textswith meaningful informations*

# Fix #[*issue_number Short description*]
[*Long description*]

# Close #[*issue_number Short description*]
[*Long description*]

# New [*Short description*]
[*Long description*]

### CODE REVIEW

File: htdocs/product/dynamic_price/class/price_parser.class.php [hunk 1]
```diff
@@ -133,7 +133,16 @@ public function translatedError()
     public function parseExpression($product, $expression, $values)
     {
         global $user;
-
+        global $hookmanager;
+        $hookmanager->initHooks(array('productcard','globalcard'));
```
Comment (eldy) [inline htdocs/product/dynamic_price/class/price_parser.class.php]: I don't think we should initialize the context of the hook here. parseExpression can be called at any place in a lot of other context.
What if you remove this line completely ?

Comment (josemariagomezroncero) [reply htdocs/product/dynamic_price/class/price_parser.class.php]: Any suggestion?

Comment (eldy) [reply htdocs/product/dynamic_price/class/price_parser.class.php]: Just remove the line $hookmanager->initHooks(array('productcard','globalcard'));

Comment (josemariagomezroncero) [reply htdocs/product/dynamic_price/class/price_parser.class.php]: there isn´t any hook loaded any times. I can add a condition
                if (!in_array('productcard', $hookmanager->hooks)) {
                    $hookmanager->initHooks(array('productcard'));
                }

Comment (eldy) [reply htdocs/product/dynamic_price/class/price_parser.class.php]: Do you want to have the hook called only when the parseExpression is called from the web page of product record or do you need it any time the parseExpression is called. I think it is the second choice, isn't it ?
You should have the context "main" that is always enabled.

Comment (josemariagomezroncero) [reply htdocs/product/dynamic_price/class/price_parser.class.php]: true, its work, thank you very much

**Pull request merged**